src/pwr.ps1

```diff
@@ -252,10 +252,17 @@ function Invoke-PullImageLayer($Out, $Repo, $Digest) {
 		}
 		Invoke-PwrWebRequest "$($Repo.Uri)/blobs/$Digest" -OutFile $Tmp -Headers $Headers
 	} else {
-		Write-PwrHost 'using cache ... ' -NoNewline
+		Write-PwrDebug "using cache for $Ref"
 	}
-	& "$env:HOMEDRIVE\Windows\System32\tar.exe" -xzf $Tmp -C $Out --exclude 'Hives/*' --strip-components 1
-	Remove-Item $Tmp
+	$Hash = "sha256:$((Get-FileHash -Path $Tmp -Algorithm SHA256).Hash)"
+	if ($Hash -ne $Digest) {
+		Write-PwrFatal "layer of $Ref with digest $Hash does not match expected digest $Digest`n     the file has been modified, implying a man-in-the-middle or similar attack"
```

Comment:
This wording is a bit alarmist. Can we just say something like `the download was corrupted` instead?

Reply:
The wording should be alarmist, but I can see why we might not want it to be. I've changed it.